[CI:DOCS] Manpage syntax proposal (#10477)
containers/podman: docs/manpage_syntax.md

% podman-command(1)

## NAME
podman\-command - short description

## SYNOPSIS
(Shows the command structure.)

> **podman command** [_optional_] _mandatory value_

Reviewer: Just checking, so these really should be `_optional_` and not `*optional*`?  We have `*optional*` all over the place.

Reviewer: or maybe   `**_optional_**`

Author: Yeah I believe you're right. Markdown applications don’t agree on how to handle underscores in the middle of a word. So in this specific case it wouldn't matter, but the the general idea I would change it to your suggestion.
But i wouldn't make it bold, since the command itself is bold.

**podman subcommand command** [_optional_] _mandatory value_

(If there is the possibility to chose between 2 (two) or more mandatory command values. There should also always be a space before and after a vertical bar to ensure better readability.)

**podman command** [_optional_] _value1_ | _value2_

**podman subcommand command** [_optional_] _value1_ | _value2_

(If an optinal value follows a mandatory one.)

**podman command** [_optional_] _value1_ | _value2_ [_optional_]

**podman subcommand command** [_optional_] _value1_ | _value2_ [_optional_]

(If the command accepts an infinite number of values.)

**podman command** [_optional_] _value_ [_value_ ...]

**podman subcommand command** [_optional_] _value_ [_value_ ...]

## DESCRIPTION

> **podman command** is always the beginning of the DESCRIPTION section. Putting the command as the first part of the decription ensures uniformity. All commands mentioned in a text retain their appearance and form.\

Reviewer: ```suggestion
**podman command** is always the beginning of the DESCRIPTION section. Putting the command as the first part of the description ensures uniformity. All commands mentioned in a text retain their appearance and form.\
```

Example sentence: The command **podman command** is an example command.\

> It should also be specified if the command can only be run as root. In addition, it should be described when a command or OPTION cannot be executed with the remote client. For a command this should be done in the DESCRIPTION part, for the OPTIONS it should be done in the DESCRIPTION of the specified OPTION.

Reviewer: Also, something to massage in here is to NOT use pronouns in the man pages, especially the word "you".

Author: I'll add that

> It should also be specified if the command can only be run as root. In addition, it should be described when a command or OPTION cannot be executed with the remote client. For a command this should be done in the DESCRIPTION part, for the OPTIONS it should be done in the DESCRIPTION of the specified OPTION.

Reviewer: ```suggestion
It should also be specified if the command can only be run as root. In addition, it should be described when a command or OPTION cannot be executed with the remote client. For a command, this should be done in the DESCRIPTION part. For the OPTIONS, it should be done in the DESCRIPTION of the specified OPTION.
```

## OPTIONS

> (All flags are referred to as OPTIONS. The term flags should not be used. All OPTIONS are listed in this section. OPTIONS that appear in descriptions of other OPTIONS and sections retain their appearance, for example: **--exit**. Each OPTION should be explained to the fullest extend below the OPTION itself. Each OPTION is behind a H4-header.)

Reviewer: ```suggestion
All flags are referred to as OPTIONS. The term flags should not be used. All OPTIONS are listed in this section. OPTIONS that appear in descriptions of other OPTIONS and sections retain their appearance, for example: **--exit**. Each OPTION should be explained to the fullest extend below the OPTION itself. Each OPTION is behind an H4-header (`####`).
```

#### **--option**, **-o**

> OPTIONS that can be put after the command in two different ways. Eather the long version with **--option** or as the short version **-o**. If there are two ways to write an OPTION they are seperated by a comma. If there are 2 (two) versions of one command the long version is always shown in front.

Reviewer: ```suggestion
OPTIONS can be put after the command in two different ways. Eather the long version with **--option** or as the short version **-o**. If there are two ways to write an OPTION they are separated by a comma. If there are 2 (two) versions of one command the long version is always shown in front.
```

#### **--exit**

> An example of a OPTION that has only one possible structure. Thus, it cannot be executed by the extention **-e**.

Reviewer: ```suggestion
An example of an OPTION that has only one possible structure. Thus, it cannot be executed by the extension **-e**.
```

#### **--answer**=, **-a**=**_active_** | _disable_

> OPTIONS that accept 2 possible arguments as inputs are shown as seen above. If there is a default argument that is selected when no special input is made, it is shown in **_bold italics_**. It must always be ensured that the standard argument is in the first place after the OPTION. In this example there are 2 (two) different versions to execute the command. Both versions of the OPTION have to be shown with the arguments behind them.

Reviewer: ```suggestion
OPTIONS that accept 2 possible arguments as inputs are shown above. If there is a default argument that is selected when no special input is made, it is shown in **_bold italics_**. It must always be ensured that the standard argument is in the first place after the OPTION. In this example, there are 2 (two) different versions to execute the command. Both versions of the OPTION have to be shown with the arguments behind them.
```

> #### **--redhat**=**good** | _better_ | _best_

Reviewer: ```suggestion
#### **--status**=**good** | _better_ | _best_
```

This is an example for 3 (three) arguments behind an OPTION. If the number of arguments is higher than 3 (three), the arguments are **not** listed after the equal sign. The arguments have to be explained in a table like in **--test**=**_test_** regardless of the number of arguments.

#### **--test**=**_test_**

OPTIONS that are followed by an equal sign include an argument after the equal sign in _italic_. If there is a default argument, that is used if the OPTION is not specified in the **command**, the argument after the eqaul sign is displayed in **bold**. All arguments must be listed and explained in the text below the OPTION.

> | argument           | description                                                                 |

Reviewer: ```suggestion
| Argument           | Description                                                                 |
```

| -                  | -                                                                           |
| **_example one_**  | This argument is the standard argument if the OPTION is not specified.      |

> | _example two_      | If one referes to a command, one should us **bold** marks.                  |

Reviewer: ```suggestion
| _example two_      | If one refers to a command, one should use **bold** marks.                  |
```

> | _example three_    | Example: In combination with **podman command** highly effectiv.            |

Reviewer: ```suggestion
| _example three_    | Example: In combination with **podman command** highly effective.            |
```

| _example four_     | Example: Can be combined with **--exit**.                                   |
| _example five_     | The fifth description                                                       |

The table shows an example for a listing of arguments. The contents in the table should be aligned left. If the content in the table conflicts with this, it can be aligned in a way that supports the understanding of the content. If there is a standard argument, it **must** listed as the first entry in the table.

> If the number of arguments is smaller than 4 (four) they have to be listed behind the OPTION as seen in the OPTION **--redhat**.

Reviewer: ```suggestion
If the number of arguments is smaller than 4 (four) they have to be listed behind the OPTION as seen in the OPTION **--status**.
```

## SUBCHAPTER
For chapters that are made specifically as an individual SUBCHAPTER in a man page, the previous conditions regarding formatting apply.

There are no restrictions for the use of paragraphs and tables. Within these paragraphs and tables the previous conditions regarding formatting apply.

Strings of characters or numbers can be highlighted with `backticks`. Paths of any kind **must** be highlighted.\
IMPORTANT: Only characters that are **not** part of categories mentioned before can be highlighted. This includes headers. For example it is not advised to highlight an OPTION or a **command**.

SUBHEADINGS are displayed as follows:
### SUBHEADING
Text for SUBHEADINGS.

## EXAMPLES

> (All EXAMPLES are listed in this section. This section should be at the end of each man page. Each EXAMPLE is always in one box. The box starts and ends with the last written line, **not** with a blank line. The `$` in front of the commands indicates that it can be run as a normal user, while the commands starting with `#` can only be run as root.)

Reviewer: You'd two of these sections that were within parens.  I'd drop the parens as I have here and up above.  However, if you feel really strongly about them, I'd be consistent in their use throughout.

```suggestion
All EXAMPLES are listed in this section. This section should be at the end of each man page. Each EXAMPLE is always in one box. The box starts and ends with the last written line, **not** with a blank line. The `$` in front of the commands indicates that it can be run as a normal user, while the commands starting with `#` can only be run as root.
```
